starport/pkg/spn/spn.go

```diff
@@ -302,6 +302,93 @@ type GenesisAccount struct {
 	Coins   types.Coins
 }
 
+// ChainSummary represents the summary of a chain in Genesis module of SPN.
+type ChainSummary struct {
+	ChainID            string
+	Source             string
+	TotalValidators    int
+	ApprovedValidators int
+	TotalProposals     int
+	ApprovedProposals  int
+}
+
+// ChainList lists chain summaries
+func (c *Client) ChainList(ctx context.Context, accountName string) ([]ChainSummary, error) {
+	clientCtx, err := c.buildClientCtx(accountName)
+	if err != nil {
+		return []ChainSummary{}, err
+	}
+
+	// List the chains
+	q := genesistypes.NewQueryClient(clientCtx)
+	req := &genesistypes.QueryListChainsRequest{}
+	chainList, err := q.ListChains(ctx, req)
+	if err != nil {
+		return []ChainSummary{}, err
+	}
+
+	var chainSummaries []ChainSummary
+
+	// Get the summary of each chain
+	for _, chain := range chainList.Chains {
+		var chainSummary ChainSummary
+		chainSummary.ChainID = chain.ChainID
+		chainSummary.Source = chain.SourceURL
+
+		// Get the number of validators
+		reqValidators := &genesistypes.QueryListProposalsRequest{
+			ChainID: chain.ChainID,
+			Status:  genesistypes.ProposalStatus_ANY_STATUS,
+			Type:    genesistypes.ProposalType_ADD_VALIDATOR,
+		}
+		resValidators, err := q.ListProposals(ctx, reqValidators)
```

Comment:
What about calling this API once without any filter to avoid extra calls?

Reply:
Yeah I was wondering about this. I find it cleaner to directly used the queries provided by the blockchain but it can bring performance issues if the command is called a lot.

I would personally track it. Since we may also implement a cache in `spnd` querier and maybe it could solve this issue as well

Reply:
I'm very curious about the cache stuff!

I think it seems proper to make a single call to `ListProposals` in order to make the code simpler and provide a -slightly- faster experience to end users. Still, we can track this by another issue to solve in a later time.